components/brave_shields/browser/brave_shields_web_contents_observer.cc

```diff
@@ -12,11 +12,15 @@
 #include <vector>
 
 #include "base/strings/utf_string_conversions.h"
+#include "brave/browser/brave_browser_process_impl.h"
 #include "brave/common/extensions/api/brave_shields.h"
 #include "brave/common/pref_names.h"
 #include "brave/common/render_messages.h"
 #include "brave/components/brave_shields/browser/brave_shields_util.h"
 #include "brave/components/brave_shields/common/brave_shield_constants.h"
+#include "brave/components/brave_site_specific/browser/site_specific_script_config_service.h"
```

Comment:
I think all of this should go into a new WebContentsUserData/WebContentsObserver in the site scripts component. Also fyi - this one is misnamed because the convention is `*TabHelper` so it would be something like `SiteScriptsTabHelper`